core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -271,31 +272,22 @@ class TransactionStateManager(brokerId: Int,
    * When this broker becomes a follower for a transaction log partition, clear out the cache for corresponding transactional ids
    * that belong to that partition.
    */
-  def removeTransactionsForPartition(partition: Int) {
+  def removeTransactionsForTxnTopicPartition(partitionId: Int) {
     validateTransactionTopicPartitionCountIsStable()
 
-    val topicPartition = new TopicPartition(Topic.TransactionStateTopicName, partition)
-
-    inLock(stateLock) {
-      ownedPartitions.remove(partition)
-      loadingPartitions.remove(partition)
-    }
+    val topicPartition = new TopicPartition(Topic.TransactionStateTopicName, partitionId)
```

Comment:
This will be in completed in another PR for the locking mechanism, and here is a sketch:

1. we will add a per-broker read-write lock on TC.
2. All request-handling logic need to grab the read lock before accessing the cache for checking state until appending to the local log call returns.
3. the loading / removing thread will grab the write lock when removing / adding the sub-map for that partition into the cache.
4. implementation-wise, we will modify `stateLock` for that read-write lock. The main purpose is to avoid log appending out-of-order, e.g. consider the following order:

```
a. check on cache passed.
b. write to local txn log

between a and b, long GC happens and emigration happens, followed by an immigration so that the broker re-become the tc for that txn partition. So step b) will succeed but there could be more entries already appended after checking step a)
```